GETTING_STARTED_CLI.markdown

```diff
@@ -0,0 +1,294 @@
+walkthrough will assume you're in a docker container capable of building
+Veracruz.
+
+## Building the Veracruz binaries
+
+First lets make sure we've built and installed the Veracruz binaries. For
+this part you need to specify which Trusted Execution Environment (TEE) you are
+using. So for example, if you are running Veracruz on SGX, you would
+run `make sgx-cli-install`, if you are using TrusteZone, you would
+run `make trustzone-cli-install`.
+
+We're also going to need a WebAssembly toolchain to build our example binary.
+This can be built using `make sdk`. Fortunately, you can combine these two
+rules, allowing you to maximize the time you have to get coffee while the code
+is compiling:
+
+``` bash
+$ make sgx-cli-install sdk
+...
+```
+
+You should now have the Veracruz binaries installed:
+
+``` bash
+$ vc-client --help
+veracruz-client 0.3.0
+
+USAGE:
+    vc-client [FLAGS] [OPTIONS] <policy-path> --identity <identity> --key <key> --target <target>
+
+...
+```
+
+Veracruz contains several independent components that are available as
+separate binaries:
+
+- `vc-pgen`/`generate-policy` - Generates and validates policy files
+  used to describe a Veracruz computation.
+- `vc-pas`/`proxy-attestation-server` - Runs the Veracruz proxy-attestation
+  server, which is used to attest Veracruz server instances.
+- `vc-server`/`veracruz-server` - Runs the Veracruz server, which provides a
+  trusted, attestable enclave for Veracruz computation.
+- `vc-client`/`veracruz-client` - Communicates with the Veracruz server using
+  an identity's certificate to upload/download data and programs for
+  computation.
+- `vc-fee`/`freestanding-execution-environment` - Provides a freestanding
+  execution environment that can be used to test Veracruz programs without
+  needing the full attestation/TEE framework.
+- `vc-wc`/`wasm-checker` - Checks that a Veracruz program is able to run in
+  a Veracruz computation.
+
+## Building the example
+
+For this walkthrough we're going to use the Shamir secret sharing example,
+which can be found in the `sdk/rust-examples/shamir-secret-sharing`
+directory.
+
+We need to use Xargo to cross-compile this example into WebAssembly, fortunately
+this is all captured in the example's Makefile:
+
+``` bash
+$ cd sdk/rust-examples/shamir-secret-sharing
+$ make
+...
+```
+
+You should now find the example compiled to WebAssembly in the `target/wasm32-wasi/release`
+directory. This will be our program to execute inside a Veracruz enclave:
+
+``` bash
+$ ls target/wasm32-wasi/release/shamir-secret-sharing.wasm
+target/wasm32-wasi/release/shamir-secret-sharing.wasm
+```
+
+Lets go ahead and copy this to an example directory to make the paths a bit
+easier to use:
+
+``` bash
+$ cd /work/veracruz
+$ mkdir example
+$ cp sdk/rust-examples/shamir-secret-sharing/target/wasm32-wasi/release/shamir-secret-sharing.wasm example/example-binary.wasm
+```
+
+<!-- TODO run example with freestanding-execution-environment? -->
```

Comment:
I wasn't planning to, the freestanding-execution-environment needs a bit of work first. It currently doesn't have options that support the enclave-fs names. Rather than adding that to this PR I was planning to make changes to the freestanding-execution-environment a later PR.

I can either 1. create an issue to be cleaned up later or 2. add it to this PR. I don't really have a preference.

Reply:
Moved here: https://github.com/veracruz-project/veracruz/issues/173